stonesoup/updater/kalman.py

```diff
@@ -280,8 +280,10 @@ def update(self, hypothesis, **kwargs):
             posterior_covariance = \
                 (posterior_covariance + posterior_covariance.T)/2
 
-        return self._update_class(posterior_mean, posterior_covariance, hypothesis,
-                                  timestamp=hypothesis.measurement.timestamp)
+        return Update.from_state(
+            hypothesis.prediction, self._update_class,
+            state_vector=posterior_mean, covar=posterior_covariance,
+            timestamp=hypothesis.measurement.timestamp, hypothesis=hypothesis)
```

Comment:
So given previous comment, the timestamp could be taken from the prediction...

Reply:
So I tried this without, but some initiators have a prior without a defined timestamp, and which causes an issue. I wonder if this could effect the `predicted_measurement` as well?